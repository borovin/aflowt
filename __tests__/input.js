const Browser = require('../tools/browser');

let browser = null;

beforeEach(() => {
    browser = new Browser();
    browser.goto('/examples/input.html');
});

afterEach(() => {
    browser.close();
});

test('Default text input', async () => {
    const snapshot = await browser.snapshot('b-input:nth-child(1)');

    expect(snapshot).toMatchSnapshot();

    browser.close();
});

test('Floating label input', async () => {
    const snapshot = await browser.snapshot('b-input:nth-child(2)');

    expect(snapshot).toMatchSnapshot();

    browser.close();
});

test('Filled input', async () => {
    const snapshot = await browser.snapshot('b-input:nth-child(3)');

    expect(snapshot).toMatchSnapshot();

    browser.close();
});

test('Input error', async () => {
    const snapshot = await browser.snapshot('b-input:nth-child(4)');

    expect(snapshot).toMatchSnapshot();

    browser.close();
});