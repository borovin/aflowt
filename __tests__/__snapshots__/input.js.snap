// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default text input 1`] = `
<b-input name="a" placeholder="Input Placeholder" type="text">
  <b-input-text placeholder="Input Placeholder" type="text">
    <input type="text" placeholder="Input Placeholder">
    <b-input-text--border></b-input-text--border>
  </b-input-text>
</b-input>
`;

exports[`Filled input 1`] = `
<b-input name="c" value="Example Value" label="Input Label" type="text">
  <b-input-text value="Example Value" label="Input Label" type="text">
    <input value="Example Value" type="text">
    <b-input-text--border></b-input-text--border>
    <b-input-text--label>Input Label</b-input-text--label>
  </b-input-text>
</b-input>
`;

exports[`Floating label input 1`] = `
<b-input name="b" label="Floating Label" type="text">
  <b-input-text label="Floating Label" type="text">
    <input type="text">
    <b-input-text--border></b-input-text--border>
    <b-input-text--label>Floating Label</b-input-text--label>
  </b-input-text>
</b-input>
`;

exports[`Input error 1`] = `
<b-input name="d" value="Example Value" label="Input Label" error="some input error"
type="text">
  <b-input-text value="Example Value" label="Input Label" error="some input error"
  type="text">
    <input value="Example Value" type="text">
    <b-input-text--border></b-input-text--border>
    <b-input-text--label>Input Label</b-input-text--label>
  </b-input-text>
</b-input>
`;
